Woodoku/src/entity/woodoku_board.py

```diff
@@ -31,20 +31,30 @@ def add_blocks(self, blocks_coord: Iterable[Tuple[int, int]]) -> None:
         """
         Mark each position specified in blocks_coord as True to indicate that the position is occupied.
 
-        :param blocks_coord: a list of (x,y) tuples to be added to the board
-        :return:
+        Args:
+             blocks_coord: a list of (x,y) tuples to be added to the board
+
+        Raises:
+            ShapeOutOfBoundError: if any block in `blocks` is invalid
         """
         for row, col in blocks_coord:
+            if not self.__validate((row, col)):
+                raise ShapeOutOfBoardError(row, col)
```

Comment:
```suggestion
            self.__validate(row, col)
```
As a simple example simply do this